src/bench/addrman.cpp

```diff
@@ -95,6 +110,43 @@ static void AddrManSelect(benchmark::Bench& bench)
     });
 }
 
+// The worst case performance of the Select() function is when there is only
+// one address on the table, because it linearly searches every position of
+// several buckets before identifying the correct bucket
+static void AddrManSelectFromAlmostEmpty(benchmark::Bench& bench)
+{
+    AddrMan addrman{EMPTY_NETGROUPMAN, /*deterministic=*/false, ADDRMAN_CONSISTENCY_CHECK_RATIO};
+
+    // Add one address to the new table
+    CService addr = ResolveService("250.3.1.1", 8333);
+    addrman.Add({CAddress(addr, NODE_NONE)}, ResolveService("250.3.1.1", 8333));
+
+    bench.run([&] {
+        const auto& address = addrman.Select();
+        assert(address.first.GetPort() > 0);
```

Comment:
Is checking the port an indirect way to check that `Select()` returned a valid result? You can check that the returned value `IsValid()` for that. But same as for the other bench, this may as well be just `(void)addrman.Select();` to reduce the noise (and leave correctness checks to the unit tests).